src/ride/ride.c

```diff
@@ -4470,7 +4470,7 @@ train_ref vehicle_create_train(int rideIndex, int x, int y, int z, int *remainin
 void vehicle_create_trains(int rideIndex, int x, int y, int z, rct_map_element *mapElement)
 {
 	rct_ride *ride = get_ride(rideIndex);
-	train_ref firstTrain, lastTrain;
+	train_ref firstTrain = { 0 }, lastTrain = { 0 };
```

Comment:
this doesn't silently suppress the warning, it initialises memory to a known value, which if for some reason gets used in line 4501/4502, will cause a crash rather then perhaps silently producing bogus results.